ConsoleUserInterface.cs

```diff
@@ -3,40 +3,58 @@
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
+using TicTacToeGame.ClassPractice;
 
 namespace TicTacToeGame;
 
 internal class ConsoleUserInterface
 {
-    public static byte TakeUserInput(byte[] xSquares, byte[] oSquares)
+    public static byte MoveInputWithErrorCheck(byte[] xSquares, byte[] oSquares)
     {
-        byte input = ConsoleInputs.GetConsoleInput();
-        while (Tools.IsInputError(input, xSquares, oSquares))
+        byte playerMove = ConsoleInputs.GetConsoleByteInput();
+        if (IsInputError(playerMove, xSquares, oSquares))
         {
-            ConsoleOutputs.DisplayError();
-            input = ConsoleInputs.GetConsoleInput();
+            ConsoleOutputs.DisplayLine(Constants.invalidInputLine);
+            playerMove = ConsoleInputs.GetConsoleByteInput();
         }
-        return input;
+        return playerMove;
     }
 
-    public static void ViewTicTacToeBoard(byte[] xSquares, byte[] oSquares)
+    public static string[] TakeUserNames()
     {
-        char[] gameState = new char[9];
-        for (byte i = 1; i < 10; i++)
+        ConsoleOutputs.DisplayLine("Player X name:");
+        string playerXName = ConsoleInputs.GetConsoleStringInput();
+
+        ConsoleOutputs.DisplayLine("Player O name:");
+        string playerOName = ConsoleInputs.GetConsoleStringInput();
+
+        return [playerXName, playerOName];
+    }
+
+    public static int ChooseAndDisplayGameHistoryFile(List<History> historyList)
+    {
+        ConsoleOutputs.DisplayLine(Constants.chooseAGameLine);
+        int chosenGame = ConsoleInputs.GetConsoleByteInput() - 1;
+
+        byte[] xMoveHistory = Tools.IntegerToByteArray(historyList[chosenGame].XMoveHistory, historyList[chosenGame].XMoveHistory.ToString().Length);
+        byte[] oMoveHistory = Tools.IntegerToByteArray(historyList[chosenGame].OMoveHistory, historyList[chosenGame].OMoveHistory.ToString().Length);
+
+        Console.Clear();
+        ConsoleOutputs.ScrollHistoryGameDisplay(xMoveHistory, oMoveHistory);
+
+        return chosenGame;
+    }
+
+    private static bool IsInputError(byte input, byte[] xSquares, byte[] oSquares)
+    {
+        if (input <= Constants.maximumNumberOfTurnsAndInputValues || input >= Constants.minimumInputValue)
+        {
+            return true;
```

Comment:
return input <= Constants.maximumNumberOfTurnsAndInputValues || input >= Constants.minimumInputValue || xSquares.Contains(input) || oSquares.Contains(input);

rename to IsInputValid
return IsValidInputNumber && IsInputContainerInSquares;